app/src/main/java/info/nightscout/androidaps/utils/stats/TddCalculator.kt

```diff
@@ -41,54 +41,25 @@ class TddCalculator @Inject constructor(
     fun calculate(days: Long): LongSparseArray<TotalDailyDose> {
         var startTime = MidnightTime.calc(dateUtil.now() - T.days(days).msecs())
         val endTime = MidnightTime.calc(dateUtil.now())
+        val stepSize = T.hours(24).msecs()
 
         val result = LongSparseArray<TotalDailyDose>()
         // Try to load cached values
         while (startTime < endTime) {
             val tdd = repository.getCalculatedTotalDailyDose(startTime).blockingGet()
             if (tdd is ValueWrapper.Existing) result.put(startTime, tdd.value)
             else break
-            startTime += T.hours(24).msecs()
+            startTime += stepSize
         }
 
         if (endTime > startTime) {
-            repository.getBolusesDataFromTimeToTime(startTime, endTime, true).blockingGet()
-                .filter { it.type != Bolus.Type.PRIMING }
-                .forEach { t ->
-                    val midnight = MidnightTime.calc(t.timestamp)
-                    val tdd = result[midnight] ?: TotalDailyDose(timestamp = midnight)
-                    tdd.bolusAmount += t.amount
-                    result.put(midnight, tdd)
-                }
-            repository.getCarbsDataFromTimeToTimeExpanded(startTime, endTime, true).blockingGet().forEach { t ->
-                val midnight = MidnightTime.calc(t.timestamp)
-                val tdd = result[midnight] ?: TotalDailyDose(timestamp = midnight)
-                tdd.carbs += t.amount
-                result.put(midnight, tdd)
-            }
-
-            val calculationStep = T.mins(5).msecs()
-            val tempBasals = iobCobCalculator.getTempBasalIncludingConvertedExtendedForRange(startTime, endTime, calculationStep)
```

Comment:
this was more efficient because it reduced amount of SQL calls significantly

Reply:
not so clear, but still an improvement: because it were burning cpu in [nested loops](https://github.com/justmara/AndroidAPS/blob/fix/tdd-cache/app/src/main/java/info/nightscout/androidaps/plugins/iob/iobCobCalculator/IobCobCalculatorPlugin.kt#L561) while continuosly searching for matches for every step in interval. not it mostly hits or populates cache entries, used by other TDD calculating parts